kernel/include/nano_internal.h

```diff
@@ -95,6 +95,30 @@ static inline unsigned int _Swap(unsigned int key)
  * @return Max number of free regions, or -1 if there is no limit
  */
 extern int _arch_mem_domain_max_partitions_get(void);
+
+/**
+ * @brief Remove a partition from the memory domain
+ *
+ * A memory domain contains multiple partitions and this API provides the
+ * freedom to remove a particular partition while keeping others intact.
+ * This API will handle any arch/HW specific changes that needs to be done.
+ *
+ * @param domain The memory domain structure
+ * @param partition_id The partition that needs to be deleted
+ */
+extern void _arch_mem_domain_remove_partition(struct k_mem_domain *domain,
```

Comment:
Zephyr coding standard is to have the verb last.
Should be _arch_mem_domain_partition_remove